File: solutions/de.vimotest.sandbox/source_gen/de/vimotest/sandbox/widgetassertions/TextBoxes_ViewTest.cpp
#include <gtest/gtest.h>
#include <memory>
#include "TextBoxes_ViewModelTestEnvironment.h"
#include <string>
#include "TextBoxes_ViewModel.h"
#include <vector>
#include "TextBoxes_ViewModelTestEnvironmentImpl.h"

namespace widgetassertions
{
  class TextBoxes_ViewTest : public testing::Test
  {
  public:
    std::shared_ptr<TextBoxes_ViewModel> sut;
    std::shared_ptr<TextBoxes_ViewModelTestEnvironment> testEnvironment;
    virtual void BuildSut();
    virtual void given_empty_context();
    virtual void then_MyValue_shows_text_Some_entered_Input_and_is_enabled_and_is_visible();
    virtual void then_MyValue_shows_text_Some_entered_Input_and_is_not_enabled_and_is_visible();
    virtual void then_MyValue_shows_text_Some_entered_Input_and_is_enabled_and_is_not_visible();
    virtual void then_MultiLineTextBoxValue_shows_text_Expected1_Expected2_();
  protected:
    void SetUp() override;
  };
  void TextBoxes_ViewTest::SetUp()
  {
    this->testEnvironment = std::make_shared<TextBoxes_ViewModelTestEnvironmentImpl>();
    this->testEnvironment->Init();
  }
  TEST_F(TextBoxes_ViewTest, SingleLine_TextBox_given_empty_context_when_then_MyValue_shows_text_Some_entered_Input_and_is_enabled_and_is_visible)
  {
    this->given_empty_context();
    this->BuildSut();
    this->then_MyValue_shows_text_Some_entered_Input_and_is_enabled_and_is_visible();
  }
  TEST_F(TextBoxes_ViewTest, SingleLine_TextBox_disabled_given_empty_context_when_then_MyValue_shows_text_Some_entered_Input_and_is_not_enabled_and_is_visible)
  {
    this->given_empty_context();
    this->BuildSut();
    this->then_MyValue_shows_text_Some_entered_Input_and_is_not_enabled_and_is_visible();
  }
  TEST_F(TextBoxes_ViewTest, SingleLine_TextBox_invisible_given_empty_context_when_then_MyValue_shows_text_Some_entered_Input_and_is_enabled_and_is_not_visible)
  {
    this->given_empty_context();
    this->BuildSut();
    this->then_MyValue_shows_text_Some_entered_Input_and_is_enabled_and_is_not_visible();
  }
  TEST_F(TextBoxes_ViewTest, MultiLine_TextBox_given_empty_context_when_then_MultiLineTextBoxValue_shows_text_Expected1_Expected2_)
  {
    this->given_empty_context();
    this->BuildSut();
    this->then_MultiLineTextBoxValue_shows_text_Expected1_Expected2_();
  }
  void TextBoxes_ViewTest::BuildSut()
  {
    this->sut = this->testEnvironment->BuildSut();
  }
  void TextBoxes_ViewTest::given_empty_context()
  {
  }
  void TextBoxes_ViewTest::then_MyValue_shows_text_Some_entered_Input_and_is_enabled_and_is_visible()
  {
    ASSERT_EQ(std::string("Some entered Input"), this->sut->getMyValueTextBoxText()) << std::string("Expected that textbox MyValue has text <") + std::string("Some entered Input") + std::string(">, but was <") + this->sut->getMyValueTextBoxText() + std::string(">");
    EXPECT_TRUE(this->sut->getIsMyValueTextBoxEnabled(), std::string("Expected textbox MyValue is enabled, but it was disabled"));
    EXPECT_TRUE(this->sut->getIsMyValueTextBoxVisible(), std::string("Expected textbox MyValue is visible, but it was hidden"));
  }
  void TextBoxes_ViewTest::then_MyValue_shows_text_Some_entered_Input_and_is_not_enabled_and_is_visible()
  {
    ASSERT_EQ(std::string("Some entered Input"), this->sut->getMyValueTextBoxText()) << std::string("Expected that textbox MyValue has text <") + std::string("Some entered Input") + std::string(">, but was <") + this->sut->getMyValueTextBoxText() + std::string(">");
    EXPECT_FALSE(this->sut->getIsMyValueTextBoxEnabled(), std::string("Expected textbox MyValue is disabled, but it was enabled"));
    EXPECT_TRUE(this->sut->getIsMyValueTextBoxVisible(), std::string("Expected textbox MyValue is visible, but it was hidden"));
  }
  void TextBoxes_ViewTest::then_MyValue_shows_text_Some_entered_Input_and_is_enabled_and_is_not_visible()
  {
    ASSERT_EQ(std::string("Some entered Input"), this->sut->getMyValueTextBoxText()) << std::string("Expected that textbox MyValue has text <") + std::string("Some entered Input") + std::string(">, but was <") + this->sut->getMyValueTextBoxText() + std::string(">");
    EXPECT_TRUE(this->sut->getIsMyValueTextBoxEnabled(), std::string("Expected textbox MyValue is enabled, but it was disabled"));
    EXPECT_FALSE(this->sut->getIsMyValueTextBoxVisible(), std::string("Expected textbox MyValue is hidden, but it was visible"));
  }
  void TextBoxes_ViewTest::then_MultiLineTextBoxValue_shows_text_Expected1_Expected2_()
  {
    auto& actualMultiLineTextBoxValueLines = this->sut->getMultiLineTextBoxValueTextBoxText();
    ASSERT_EQ(2, actualMultiLineTextBoxValueLines.size()) << std::string("Expected that textbox MultiLineTextBoxValue has text with 2 lines, but has ") + actualMultiLineTextBoxValueLines.size();
    ASSERT_EQ(std::string("Expected1"), actualMultiLineTextBoxValueLines.at(0)) << std::string("Expected that textbox MultiLineTextBoxValue has text line at index 0 matching <") + std::string("Expected1") + std::string(">, but was <") + actualMultiLineTextBoxValueLines.at(0) + std::string(">");
    ASSERT_EQ(std::string("Expected2"), actualMultiLineTextBoxValueLines.at(1)) << std::string("Expected that textbox MultiLineTextBoxValue has text line at index 1 matching <") + std::string("Expected2") + std::string(">, but was <") + actualMultiLineTextBoxValueLines.at(1) + std::string(">");
  }
}

File: solutions/de.vimotest.sandbox/source_gen/de/vimotest/sandbox/widgetassertions/TextBoxes_ViewTest.cs
/// <filename>
///     TextBoxes_ViewTest.cs
/// </filename>
using widgetassertions;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.Collections.Generic;
using System;

[Microsoft.VisualStudio.TestTools.UnitTesting.TestClass]
public class TextBoxes_ViewTest
{
    private widgetassertions.TextBoxes_ViewModel sut;

    private widgetassertions.TextBoxes_ViewModelTestEnvironment testEnvironment;

    [Microsoft.VisualStudio.TestTools.UnitTesting.TestInitialize]
    public void SetUp()
    {
        this.testEnvironment = new widgetassertions.TextBoxes_ViewModelTestEnvironmentImpl();
        this.testEnvironment.Init();
    }

    /*

    Scenario: SingleLine TextBox
      given: empty context
       when:
       then: MyValue shows text 'Some entered Input' and is enabled and is visible
   */
    [Microsoft.VisualStudio.TestTools.UnitTesting.TestMethod]
    public void SingleLine_TextBox_given_empty_context_when_then_MyValue_shows_text_Some_entered_Input_and_is_enabled_and_is_visible()
    {
        this.given_empty_context();
        this.BuildSut();
        this.then_MyValue_shows_text_Some_entered_Input_and_is_enabled_and_is_visible();
    }

    /*

    Scenario: SingleLine TextBox (disabled)
      given: empty context
       when:
       then: MyValue shows text 'Some entered Input' and is not enabled and is visible
   */
    [Microsoft.VisualStudio.TestTools.UnitTesting.TestMethod]
    public void SingleLine_TextBox_disabled_given_empty_context_when_then_MyValue_shows_text_Some_entered_Input_and_is_not_enabled_and_is_visible()
    {
        this.given_empty_context();
        this.BuildSut();
        this.then_MyValue_shows_text_Some_entered_Input_and_is_not_enabled_and_is_visible();
    }

    /*

    Scenario: SingleLine TextBox (invisible)
      given: empty context
       when:
       then: MyValue shows text 'Some entered Input' and is enabled and is not visible
   */
    [Microsoft.VisualStudio.TestTools.UnitTesting.TestMethod]
    public void SingleLine_TextBox_invisible_given_empty_context_when_then_MyValue_shows_text_Some_entered_Input_and_is_enabled_and_is_not_visible()
    {
        this.given_empty_context();
        this.BuildSut();
        this.then_MyValue_shows_text_Some_entered_Input_and_is_enabled_and_is_not_visible();
    }

    /*

    Scenario: MultiLine TextBox
      given: empty context
       when:
       then: MultiLineTextBoxValue shows text 'Expected1|Expected2'
   */
    [Microsoft.VisualStudio.TestTools.UnitTesting.TestMethod]
    public void MultiLine_TextBox_given_empty_context_when_then_MultiLineTextBoxValue_shows_text_Expected1_Expected2_()
    {
        this.given_empty_context();
        this.BuildSut();
        this.then_MultiLineTextBoxValue_shows_text_Expected1_Expected2_();
    }

    protected virtual void BuildSut()
    {
        this.sut = this.testEnvironment.BuildSut();
    }

    // Given Helper Definitions
    public virtual void given_empty_context()
    {
    }

    // Then Helper Definitions
    public virtual void then_MyValue_shows_text_Some_entered_Input_and_is_enabled_and_is_visible()
    {
        Assert.AreEqual("Some entered Input", this.sut.getMyValueTextBoxText(), "Expected that textbox MyValue has text <" + "Some entered Input" + ">, but was <" + this.sut.getMyValueTextBoxText() + ">");
        Assert.IsTrue(this.sut.getIsMyValueTextBoxEnabled(), "Expected textbox MyValue is enabled, but it was disabled");
        Assert.IsTrue(this.sut.getIsMyValueTextBoxVisible(), "Expected textbox MyValue is visible, but it was hidden");
    }

    public virtual void then_MyValue_shows_text_Some_entered_Input_and_is_not_enabled_and_is_visible()
    {
        Assert.AreEqual("Some entered Input", this.sut.getMyValueTextBoxText(), "Expected that textbox MyValue has text <" + "Some entered Input" + ">, but was <" + this.sut.getMyValueTextBoxText() + ">");
        Assert.IsFalse(this.sut.getIsMyValueTextBoxEnabled(), "Expected textbox MyValue is disabled, but it was enabled");
        Assert.IsTrue(this.sut.getIsMyValueTextBoxVisible(), "Expected textbox MyValue is visible, but it was hidden");
    }

    public virtual void then_MyValue_shows_text_Some_entered_Input_and_is_enabled_and_is_not_visible()
    {
        Assert.AreEqual("Some entered Input", this.sut.getMyValueTextBoxText(), "Expected that textbox MyValue has text <" + "Some entered Input" + ">, but was <" + this.sut.getMyValueTextBoxText() + ">");
        Assert.IsTrue(this.sut.getIsMyValueTextBoxEnabled(), "Expected textbox MyValue is enabled, but it was disabled");
        Assert.IsFalse(this.sut.getIsMyValueTextBoxVisible(), "Expected textbox MyValue is hidden, but it was visible");
    }

    public virtual void then_MultiLineTextBoxValue_shows_text_Expected1_Expected2_()
    {
        var actualMultiLineTextBoxValueLines = this.sut.getMultiLineTextBoxValueTextBoxText();
        Assert.AreEqual(2, actualMultiLineTextBoxValueLines.Count, "Expected that textbox MultiLineTextBoxValue has text with 2 lines, but has " + Convert.ToString(actualMultiLineTextBoxValueLines.Count));
        Assert.AreEqual("Expected1", actualMultiLineTextBoxValueLines[0], "Expected that textbox MultiLineTextBoxValue has text line at index 0 matching <" + "Expected1" + ">, but was <" + actualMultiLineTextBoxValueLines[0] + ">");
        Assert.AreEqual("Expected2", actualMultiLineTextBoxValueLines[1], "Expected that textbox MultiLineTextBoxValue has text line at index 1 matching <" + "Expected2" + ">, but was <" + actualMultiLineTextBoxValueLines[1] + ">");
    }
}
